worker/default_lifecycle_test.go

```diff
@@ -0,0 +1,127 @@
+package worker_test
+
+import (
+	"testing"
+
+	"github.com/WatchBeam/redutil/conn"
+	"github.com/WatchBeam/redutil/queue"
+	"github.com/WatchBeam/redutil/test"
+	"github.com/WatchBeam/redutil/worker"
+	"github.com/garyburd/redigo/redis"
+	"github.com/stretchr/testify/suite"
+)
+
+type DefaultLifecycleSuite struct {
+	*test.RedisSuite
+}
+
+func TestDefaultLifecycleSuite(t *testing.T) {
+	pool, _ := conn.New(conn.ConnectionParam{
+		Address: "127.0.0.1:6379",
+	}, 1)
+
+	suite.Run(t, &DefaultLifecycleSuite{test.NewSuite(pool)})
+}
+
+func (suite *DefaultLifecycleSuite) TestConstruction() {
+	l := worker.NewLifecycle(suite.Pool, "queue", "worker_1")
+
+	suite.Assert().IsType(&worker.DefaultLifecycle{}, l)
+}
+
+func (suite *DefaultLifecycleSuite) TestListenReturnsTasks() {
+	l := worker.NewLifecycle(suite.Pool, "queue", "worker_1")
+	queue := queue.NewByteQueue(suite.Pool, "queue")
+	tasks, _ := l.Listen()
+	defer func() {
+		l.AbandonAll()
+	}()
+
+	queue.Push([]byte("hello, world!"))
+
+	task := <-tasks
+	l.StopListening()
+
+	suite.Assert().Equal([]byte("hello, world!"), task.Bytes())
+}
+
+func (suite *DefaultLifecycleSuite) TestCompletedTasksRemovedFromAllQueues() {
+	l := worker.NewLifecycle(suite.Pool, "queue", "worker_1")
+	defer l.StopListening()
+
+	queue := queue.NewByteQueue(suite.Pool, "queue")
+	queue.Push([]byte("some_task"))
+
+	tasks, _ := l.Listen()
+
+	l.Complete(<-tasks)
+
+	suite.WithRedis(func(conn redis.Conn) {
+		ql := suite.RedisLength("queue")
+		wl := suite.RedisLength("queue:_worker_worker_1")
+
+		suite.Assert().Equal(0, ql, "redutil: main queue should be empty, but wasn't")
+		suite.Assert().Equal(0, wl, "redutil: worker (worker_1) queue should be empty, but wasn't")
+	})
+}
```

Comment:
It'd also be good to have a test for the situation where concat fails part of the way through.